streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -380,7 +380,7 @@ public void testMaybePunctuateStreamTime() {
         assertFalse(task.process());
         assertFalse(task.maybePunctuateStreamTime());
 
-        processorStreamTime.supplier.checkAndClearPunctuateResult(PunctuationType.STREAM_TIME, 20L, 30L, 40L);
+        processorStreamTime.supplier.checkAndClearPunctuateResult(PunctuationType.STREAM_TIME, 20L, 32L, 40L);
```

Comment:
Punctuation is data driven, that's why we punctuate at 32, but we stay aligned as illustrated by the next punctuation at 40